service/worker/archiver/util.go

```diff
@@ -40,7 +40,7 @@ func MaxArchivalIterationTimeout() time.Duration {
 
 func hash(i interface{}) uint64 {
 	var b bytes.Buffer
-	gob.NewEncoder(&b).Encode(i)
+	_ = gob.NewEncoder(&b).Encode(i)
```

Comment:
@yycptt can you review - I think we should actually be returning the error in this case?

Reply:
Maybe. But I am not sure how caller should handle this error. This error should not cause archival to fail, so maybe just logger this error. But if that's the case, our sanity check on `hashEqual` will also fail, and we can still get a log or metrics. So I am ok with just return some default value (like 0), when the encoding failed.